src/Features/CSharp/Portable/UseRecursivePatterns/CSharpUseRecursivePatternsCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseRecursivePatterns
+{
+    internal sealed partial class CSharpUseRecursivePatternsCodeRefactoringProvider
+    {
+        private sealed class Analyzer : CSharpSyntaxVisitor<AnalyzedNode>
+        {
+            private readonly SemanticModel _semanticModel;
+
+            private Analyzer(SemanticModel semanticModel)
+            {
+                _semanticModel = semanticModel;
+            }
+
+            public static AnalyzedNode Analyze(SyntaxNode node, SemanticModel semanticModel)
+            {
+                return new Analyzer(semanticModel).Visit(node);
+            }
+
+            public override AnalyzedNode VisitCasePatternSwitchLabel(CasePatternSwitchLabelSyntax node)
+            {
+                // Attempt to combine the pattern on the left and the condition of the when-clause.
+                if (node.WhenClause is var whenClause && whenClause != null)
```

Comment:
this 'var whenClause' seems super unnecessary.  You already have multiple statement.  either replace the `whenClause`s with `node.WhenClause` or just use a local.